envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -174,20 +174,30 @@ class EnvoySnapshotFactory(
         group: Group,
         globalSnapshot: GlobalSnapshot
     ): Collection<RouteSpecification> {
+        val definedServicesRoutes = group.proxySettings.outgoing.serviceDependencies.map {
+            RouteSpecification(
+                clusterName = it.service,
+                routeDomain = it.service,
+                settings = it.settings
+            )
+        }
         return when (group) {
-            is ServicesGroup -> group.proxySettings.outgoing.getServiceDependencies().map {
-                RouteSpecification(
-                    clusterName = it.service,
-                    routeDomain = it.service,
-                    settings = it.settings
-                )
+            is ServicesGroup -> {
+                definedServicesRoutes
             }
-            is AllServicesGroup -> globalSnapshot.allServicesNames.map {
-                RouteSpecification(
-                    clusterName = it,
-                    routeDomain = it,
-                    settings = defaultDependencySettings
-                )
+            is AllServicesGroup -> {
+// TODO refactor - getWildcardServiceDependency cannot be null
```

Comment:
What about this comment?

Reply:
I want to show that getWildcardServiceDependency will never be null for AllServiceGroup. I refactored code and know this problem is obsolete